### ISSUE TITLE
Storage Page in Supabase Studio Uses Incorrect API Endpoint (/api/storage//buckets/...), Causing 404 Errors

### ISSUE THREAD

**Issue by tsensei** [supabase/supabase#30227]
# Bug Report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the Bug

When accessing the **Storage** tab in my self-hosted Supabase Studio, the application incorrectly requests bucket objects using an invalid API endpoint, resulting in a `404 Not Found` error. This issue arose after redeploying my Supabase stack today, whereas it functioned correctly two weeks ago.

## To Reproduce

1. Navigate to the **Storage** tab in Supabase Studio.
2. Observe the initial request to fetch all buckets:
    ```
    http GET https://supabase.xyz.com/api/storage/default/buckets \
      Authorization:'Basic YXN...' \
      User-Agent:insomnia/10.0.0
    ```
3. Select a specific bucket (e.g., `documents`).
4. Observe the subsequent request to list objects within the selected bucket:
    ```
    http POST https://supabase.xyz.com/api/storage//buckets/documents/objects/list \
      Authorization:'Basic YXN...' \
      User-Agent:insomnia/10.0.0
    ```
5. Notice the `404 Not Found` error response.
6. Manually modify the request URL to include `v1`:
    ```
    http POST https://supabase.xyz.com/api/storage/v1/buckets/documents/objects/list \
      Authorization:'Basic YXN...' \
      User-Agent:insomnia/10.0.0
    ```
7. Observe that the modified request successfully returns the list of objects.

## Expected Behavior

When selecting a bucket in Supabase Studio's Storage tab, the application should correctly form the API request to list objects using the `v1` endpoint, such as:

```
http POST https://supabase.xyz.com/api/storage/v1/buckets/documents/objects/list
Authorization:'Basic YXN...'
User-Agent
/10.0.0
```

This should return the list of objects without requiring manual URL adjustments.

## Screenshots

*No screenshots available.*

## System Information

- **Supabase Studio Version:** Latest deployment as of Nov 1, 2024

## Additional Context

- The issue began after redeploying the Supabase stack today, following a successful deployment two weeks prior without this problem.
- There are double slashes in the problematic API endpoint URL: `/api/storage//buckets/documents/objects/list`.
- Manually correcting the URL to include `v1` resolves the issue, suggesting a potential misconfiguration or bug in the Studio's API request formation.
- This might be related to changes in the Supabase API routing or Studio application logic introduced in recent updates.

**Comment by antonsarg** [comment]
I can confirm that I'm also encountering this exact issue with the incorrect Storage API endpoint, resulting in a 404 error. Could you share more details on how you manually corrected the URL to make it work until this issue gets fixed?

**Comment by tsensei** [comment]
@antonsarg We downgraded to a studio image `supabase/studio:20241021-9f9b083_amd64`, all the images after this seems to have this fault, also the `/settings` endpoint that's called internally don't work too for example - which in turn not allowing the send invites etc.

After downgrading it seems fine, but maybe losing out on some dashboard features which aren't much important to us.

**Comment by antonsarg** [comment]
@tsensei Meaning that you setup your own docker-compose.yml? Feel free to write me an email, would be happy about it.

**Comment by antonsarg** [comment]
It seems that their latest beta release 1.217.2 has fixed the issue with the `/settings` endpoint. Unfortunately the storage API call has not been touched. I have downgraded to version 1.210.0 for now.

**Comment by nkmnz** [comment]
I've tried both `1.217.2` beta and downgrading to various random cli versions down to `1.172.2.` – to no avail. Any other suggestions what else I could do?

**Comment by antonsarg** [comment]
@nkmnz Make sure to delete the currently installed studio image (must be 20241029-46e1e40) before running `supabase start` after downgrading.

**Comment by antonsarg** [comment]
@tsensei Issue is fixed in version 1.219.0, I think you can consider closing this issue.

**Comment by franciscomfcmaia** [comment]
Using the latest image ([20250130-b048539](https://hub.docker.com/layers/supabase/studio/20250130-b048539/images/sha256-6802d1c6b15de3262a0633b5d0b6eeb139a062ff21eee5d674ddd3b5737da43e)) brings back this issue. Re-opening.
